test/org/zaproxy/zap/extension/pscanrulesBeta/ServletParameterPollutionScannerUnitTest.java

```diff
@@ -0,0 +1,120 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.pscanrulesBeta;
+
+import org.junit.Test;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ServletParameterPollutionScannerUnitTest extends PassiveScannerTest {
+
+    public static final String URI = "http://www.example.com/test/";
+
+    @Override
+    protected ServletParameterPollutionScanner createScanner() {
+        return new ServletParameterPollutionScanner();
+    }
+
+    @Test
+    public void GivenNoForms_WhenScanHttpResponseReceive_ThenNoAlertsRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("");
+        scanHttpResponseReceive(msg);
+        assertNumberOfAlertsRaised(0);
+    }
+
+    @Test
+    public void GivenFormWithActionAttribute_WhenScanHttpResponseReceive_ThenNoAlertsRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form action='ActionMan'>");
+        scanHttpResponseReceive(msg);
+        assertNumberOfAlertsRaised(0);
+    }
+
+    @Test
+    public void GivenFormWithNoActionAttribute_WhenScanHttpResponseReceive_ThenAlertRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form />");
+        scanHttpResponseReceive(msg);
+        assertNumberOfAlertsRaised(1);
+    }
+
+    @Test
+    public void GivenFormWithValuelessActionAttribute_WhenScanHttpResponseReceive_ThenAlertRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form action />");
+        scanHttpResponseReceive(msg);
+        assertNumberOfAlertsRaised(1);
+    }
+
+    @Test
+    public void GivenFormWithEmptyActionAttribute_WhenScanHttpResponseReceive_ThenAlertRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form action='' />");
+        scanHttpResponseReceive(msg);
+        assertNumberOfAlertsRaised(1);
+    }
+
+    @Test
+    public void GivenFormWithEmptyAndPopulatedActionAttributes_WhenScanHttpResponseReceive_ThenAlertRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form action action='ActionMan' />");
+        scanHttpResponseReceive(msg);
+        assertNumberOfAlertsRaised(1);
+    }
+
+    @Test
+    public void GivenTwoFormsWithNoActionAttribute_WhenScanHttpResponseReceive_ThenOnlyOneAlertRaised() throws Exception {
+        HttpMessage msg = createHttpMessageFromHtml("<form /><form />");
+        scanHttpResponseReceive(msg);
+        assertNumberOfAlertsRaised(1);
+    }
+
+    @Test
+    public void GivenFormWithNoActionAttribute_WhenScanHttpResponseReceive_ThenAlertPopulated() throws Exception {
```

Comment:
I'd suggest combine `GivenFormWithNoActionAttribute_WhenScanHttpResponseReceive_ThenAlertRaised()` and this one into a single test. (Then there's an alert, with the expected details...)